### PR TITLE
remove redundant closures (clippy::redundant_closure)

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: compiler/rustc_codegen_ssa/src/coverageinfo/map.rs [hunk 1]
```diff
@@ -170,30 +170,30 @@ impl<'tcx> FunctionCoverage<'tcx> {
...
-                    let index = self.expression_index(u32::from(id));
-                    self.expressions
-                        .get(index)
-                        .expect("expression id is out of range")
-                        .as_ref()
-                        // If an expression was optimized out, assume it would have produced a count
-                        // of zero. This ensures that expressions dependent on optimized-out
-                        // expressions are still valid.
-                        .map_or(Some(Counter::zero()), |_| {
-                            new_indexes[index].map(|new_index| Counter::expression(new_index))
```
Comment (matthiaskrgr) [inline compiler/rustc_codegen_ssa/src/coverageinfo/map.rs:193]: The actual change seems to be here but it caused a lot of other code to shift around :/

Comment (estebank) [reply compiler/rustc_codegen_ssa/src/coverageinfo/map.rs:193]: https://github.com/rust-lang/rust/pull/80499/files?diff=split&w=1 helps :)

**Pull request merged**